### ISSUE TITLE
Argument of type '{ link: ApolloLink; cache: InMemoryCache; defaultOptions: ...' is not assignable to parameter of type 'ApolloClientOptions<NormalizedCacheObject>'.

### ISSUE THREAD

**Issue by maapteh** [apollographql/apollo-angular#659]
I updated all dependencies to latest and have the following component:

```
    apollo.create({
      link: newLink,
      cache: new InMemoryCache(),
      defaultOptions: {
        mutate: {
          errorPolicy: 'all'
        },
        query: {
          errorPolicy: 'all'
        },
        watchQuery: {
          errorPolicy: 'all'
        }
      }
    });
```

which now gives the following error:

`
ERROR in src/app/core/graphql/core.graphql.module.ts(46,19): error TS2345: Argument of type '{ link: ApolloLink; cache: InMemoryCache; defaultOptions: { mutate: { errorPolicy: "all"; }; quer...' is not assignable to parameter of type 'ApolloClientOptions<NormalizedCacheObject>'.
  Types of property 'defaultOptions' are incompatible.
    Type '{ mutate: { errorPolicy: "all"; }; query: { errorPolicy: "all"; }; watchQuery: { errorPolicy: "al...' is not assignable to type 'DefaultOptions'.
      Types of property 'query' are incompatible.
        Type '{ errorPolicy: "all"; }' is not assignable to type 'QueryOptions<OperationVariables>'.
          Property 'query' is missing in type '{ errorPolicy: "all"; }'.
`

which i bypassed using:

```
    apollo.create({
      link: newLink,
      cache: new InMemoryCache(),
      defaultOptions: {
        mutate: {
          errorPolicy: 'all'
        },
        query: {
          errorPolicy: 'all'
        },
        watchQuery: {
          errorPolicy: 'all'
        }
      } as any
    });
```

But maybe you are able to solve the real error :)

using:
```
    "apollo-angular": "1.1.0",
    "apollo-angular-link-http": "1.1.0",
    "apollo-cache-inmemory": "1.2.3",
    "apollo-client": "2.3.3",
    "apollo-link": "1.2.2",
    "apollo-link-error": "1.1.0",
    "apollo-link-http": "1.5.4",
    "rxjs": "6.2.1",

    "@types/node": "10.0.4",
```

**Comment by hwillson** [comment]
PR #660 should help with this - thanks!

**Comment by maapteh** [comment]
tried it and works correctly with it, thanks and i will wait for the release. Apollo client is not really following semver rules.

**Comment by maapteh** [comment]
superfast! thank you!!!

**Comment by kamilkisiela** [comment]
Released `v1.1.1` with a fix
